kratos/utilities/intersection_utilities.h

```diff
@@ -143,25 +143,101 @@ class IntersectionUtilities
         rIntersectionPoint = rLinePoint1 + r*dir;
 
         // Check if the intersection point is inside the triangle
-        const double uu = inner_prod(u,u);
-        const double uv = inner_prod(u,v);
-        const double vv = inner_prod(v,v);
-        const array_1d<double,3> w = rIntersectionPoint - rTriangleGeometry[0];
-        const double wu = inner_prod(w,u);
-        const double wv = inner_prod(w,v);
-        const double D = uv * uv - uu * vv;
-
-        // Get and test parametric coords
-        const double s = (uv * wv - vv * wu) / D;
-        if (s < 0.0 - epsilon || s > 1.0 + epsilon){
-            return 0;    // Intersection point is outside the triangle
-        }
-        const double t = (uv * wu - uu * wv) / D;
-        if (t < 0.0 - epsilon || (s + t) > 1.0 + epsilon){
-            return 0;    // Intersection point is outside the triangle
+        if (PointInTriangle(rTriangleGeometry[0], rTriangleGeometry[1], rTriangleGeometry[2], rIntersectionPoint)) {
+            return 1;
         }
+        return 0;
+    }
+
+    /**
+     * Find the 2D intersection of a line (bounded) with a triangle (bounded)
+     * @param rTriangle Is the triangle to intersect
+     * @param rPoint0 Coordinates of the first point of the intersecting line
+     * @param rPoint1 Coordinates of the second point of the intersecting line
+     * @return If there is intersection
+     */
+    template <class TGeometryType>
+    static bool TriangleLineIntersection2D(
+        const TGeometryType& rTriangle,
+        const array_1d<double,3>& rPoint0,
+        const array_1d<double,3>& rPoint1)
+    {
+        return TriangleLineIntersection2D(rTriangle[0], rTriangle[1], rTriangle[2], rPoint0, rPoint1);
+    }
+
+    /**
+     * Find the 2D intersection of a line (bounded) with a triangle (bounded)
+     * @param rVert1 The first vertex of the triangle to intersect
+     * @param rVert2 The second vertex of the triangle to intersect
+     * @param rVert3 The third vertex of the triangle to intersect
+     * @param rPoint1 Coordinates of the first point of the intersecting line
+     * @param rPoint2 Coordinates of the second point of the intersecting line
+     * @return If there is intersection
+     */
+    static bool TriangleLineIntersection2D(
+        const array_1d<double,3>& rVert0,
+        const array_1d<double,3>& rVert1,
+        const array_1d<double,3>& rVert2,
+        const array_1d<double,3>& rPoint0,
+        const array_1d<double,3>& rPoint1)
+    {
+
+        // Check the intersection of each edge against the intersecting object
+        array_1d<double,3> int_point;
+        int int_id;
+        int_id = ComputeLineLineIntersection(rVert0, rVert1, rPoint0, rPoint1, int_point);
+        if (int_id) return true;
```

Comment:
```suggestion
        if (ComputeLineLineIntersection(rVert0, rVert1, rPoint0, rPoint1, int_point)) return true;
```
(same below)

Reply:
I messed up something applying that suggestion to more places. Now its fixed.